frontend/ui/widget/textviewer.lua

```diff
@@ -0,0 +1,238 @@
+        bold = true,
+        width = self.width - 2*self.title_padding - 2*self.title_margin,
+    }
+    local titlew = FrameContainer:new{
+        padding = self.title_padding,
+        margin = self.title_margin,
+        bordersize = 0,
+        CenterContainer:new{
+            dimen = Geom:new{
+                w = self.width,
+                h = title_text:getSize().h,
+            },
+            title_text
+        }
+    }
+    titlew = OverlapGroup:new{
+        dimen = {
+            w = self.width,
+            h = titlew:getSize().h
+        },
+        titlew,
+        CloseButton:new{ window = self, },
+    }
+
+    local separator = LineWidget:new{
+        dimen = Geom:new{
+            w = self.width,
+            h = Screen:scaleBySize(2),
+        }
+    }
+
+    local buttons = {
+        {
+            {
+                text = _("Close"),
+                callback = function()
+                    UIManager:close(self)
+                end,
+            },
+        },
+    }
+    local button_table = ButtonTable:new{
+        width = self.width - self.button_padding,
+        button_font_face = "cfont",
+        button_font_size = 20,
+        buttons = buttons,
+        zero_sep = true,
+        show_parent = self,
+    }
+
+    local textw_height = self.height - titlew:getSize().h - separator:getSize().h - button_table:getSize().h
+
+    self.scroll_text_w = ScrollTextWidget:new{
+            text = self.text,
+            face = self.text_face,
+            width = self.width - 2*self.text_padding - 2*self.text_margin,
+            height = textw_height - 2*self.text_padding -2*self.text_margin,
+            dialog = self,
+            justified = true,
+    }
+    local textw = FrameContainer:new{
+        padding = self.text_padding,
+        margin = self.text_margin,
+        bordersize = 0,
+        self.scroll_text_w
+    }
+
+    self.frame = FrameContainer:new{
+        radius = 8,
+        bordersize = 3,
+        padding = 0,
+        margin = 0,
+        background = Blitbuffer.COLOR_WHITE,
+        VerticalGroup:new{
+            align = "left",
+            titlew,
+            separator,
+            CenterContainer:new{
+                dimen = Geom:new{
+                    w = self.width,
+                    h = textw:getSize().h,
+                },
+                textw,
+            },
+            CenterContainer:new{
+                dimen = Geom:new{
+                    w = self.width,
+                    h = button_table:getSize().h,
+                },
+                button_table,
+            }
+        }
+    }
+    self[1] = WidgetContainer:new{
+        align = self.align,
+        dimen = self.region,
+        self.frame,
+    }
+    UIManager:setDirty("all", function()
+        local update_region = self.frame.dimen:combine(orig_dimen)
+        logger.dbg("update region", update_region)
+        return "partial", update_region
+    end)
+end
+
+function TextViewer:onCloseWidget()
```

Comment:
I tried it, and it indeed works as well without it.
This `:onCloseWidget()` is in 30% of the widgets, so it came in mine from the widget I choose as a base for this new one (dictquicklookup).
I can remove it, unless I miss a non-obvious need for it that would explain why it is there in some existing widgets.